Implemented 1D and 2D mesh tallies

samuelshaner opened the pull request:
This PR implements 1D mesh tallies for all valid scores and 2D mesh tallies for surface currents (which previously was not working). This fixes issue #63. 

Mesh filter tests have been updated to tally the surface currents since this score is implemented differently than other scores. The test results for mesh filters have also been hashed since they were getting quite long.

paulromano commented on src/mesh.F90:
```diff
@@ -248,6 +232,47 @@ end subroutine count_bank_sites
+
+    real(8) :: x0   ! track start point
+    real(8) :: x1   ! track end point
+    real(8) :: xi   ! track intersection point with mesh
```
Unused variable

samuelshaner replied:
removed.

paulromano commented on src/output.F90:
```diff
@@ -974,135 +975,147 @@ subroutine write_surface_current(t, unit_tally)
-      len1 = len_trim(string)
-      do j = 1, m % dimension(2)
-        string = string(1:len1+1) // trim(to_str(j)) // ", "
-        len2 = len_trim(string)
```
The `len1` and `len2` variables are now unused.

samuelshaner replied:
removed

paulromano commented on src/tally.F90:
```diff
@@ -2478,7 +2498,7 @@ subroutine score_surface_current(p)
 
               ! Outward current on d1 min surface
               if (all(ijk0 >= 1) .and. all(ijk0 <= m % dimension)) then
```
This line looks to be problematic because `ijk0` is always size 3 but `m % dimension` could be less. When I try to run the 1D and 2D mesh filter tests, they both fail on this line.

paulromano commented:
Hmm, this new review thing is pretty nifty!

samuelshaner commented:
Fixed that issue and now all the tests are passing. I'll change the mesh filter tests to use the python api and then this should be good to go.

samuelshaner commented:
I merged all the mesh filter tests into one test used the newly created `HashedPyAPITestingHarness` to create the test. All tests are passing and I think I have addressed all the comments.
